Sources/MongoSwift/ChangeStreamOptions.swift

```diff
@@ -1,29 +1,18 @@
 /// Describes the modes for configuring the fullDocument field of a change stream document.
-public enum FullDocument: RawRepresentable, Codable {
+public struct FullDocument: RawRepresentable, Codable {
```

Comment:
this is another case where I'm questioning the value of `RawRepresentable`. I think we could do the same thing as subtype here and get rid of it, and just allow construction via the properties and the `other` method. WDYT @patrickfreed ?

Reply:
I guess the one benefit it gets us here is that the `Codable` conformance automatically operates just on the raw value. It also sort of signals that this struct really is just representing a string option.

On the flip side, the initializer being failable may indicate that we're doing some sort of validation, which we aren't.

I would say that I don't really have strong feelings either way and am happy with whatever you guys decide.

Reply:
yeah, the failable initializer is what is bothering me 🤔 I guess we could leave it and just document that it never returns nil.